src/Adapter/Product/Combination/Update/CombinationRemover.php

```diff
@@ -48,16 +52,45 @@ class CombinationRemover
      */
     private $combinationRepository;
 
+    /**
+     * @var DefaultCombinationUpdater
+     */
+    private $defaultCombinationUpdater;
+
     /**
      * @param ProductRepository $productRepository
      * @param CombinationRepository $combinationRepository
+     * @param DefaultCombinationUpdater $defaultCombinationUpdater
      */
     public function __construct(
         ProductRepository $productRepository,
-        CombinationRepository $combinationRepository
+        CombinationRepository $combinationRepository,
+        DefaultCombinationUpdater $defaultCombinationUpdater
     ) {
         $this->productRepository = $productRepository;
         $this->combinationRepository = $combinationRepository;
+        $this->defaultCombinationUpdater = $defaultCombinationUpdater;
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     *
+     * @throws CoreException
```

Comment:
I dont know if all the `throws` are needed

Reply:
`get` throws exception,
`delete` too, 
`findDefaultCombination` and  ` setDefaultCombination` and `CombinationId` too :sweat_smile: 

This is why phpstorm adds these blocks.
But I'm still convinced that adding throws without explanation or without real throws in methods are useless.
If you change the exception inside methods, does it automatically update phpdocs for everyone? Or phpstan check it too? 

We can't rely on phpstorm configuration